src/rateEngine/DemandProfile.ts

```diff
@@ -0,0 +1,112 @@
+import { groupBy, mean, sum, times } from 'lodash';
+import LoadProfile from './LoadProfile';
+import { AveragingDemandPeriod, DemandArgs, DemandPeriod } from './types';
+import { convertInfinity } from './utils/convertInfinities';
+
+class DemandProfile {
+  private _loadProfile: LoadProfile;
+  private _demandPeriod: DemandPeriod;
+  private _averagingPeriod: AveragingDemandPeriod | undefined;
+  private _averagingQty: number | undefined;
+
+  private _min: number;
+  private _max: number;
+
+  constructor(
+    { demandPeriod = 'monthly', averagingPeriod, averagingQty, min, max, ...filters }: DemandArgs,
+    loadProfile: LoadProfile,
+  ) {
+    this.validatePeriods(averagingPeriod, demandPeriod);
+
+    this._loadProfile = loadProfile.filterBy(filters ?? {});
+    this._demandPeriod = demandPeriod;
+    this._averagingPeriod = averagingPeriod;
+    this._averagingQty = averagingQty;
+
+    this._min = min !== undefined ? convertInfinity(min) : 0;
```

Comment:
@bknoles I made the default min `0` rather than `-Infinity`, because if the user doesn't pass a min/max we do `kw - min`, which would always lead to `Infinity`.

Do we ever have to consider negative loads here? If not, I think `0` is the way to go

Reply:
Net metered load profiles could definitely have negative loads, though I don't know if we've actually encountered "tiers + net metered load profile" in the wild. Maybe we shouldn't use defaults and instead add the guard clause you had back in there. It'd also be more performant when min/max isn't defined, now that I think about it.